yt/frontends/open_pmd/data_structures.py

```diff
@@ -402,29 +435,47 @@ def _set_paths(self, handle, path):
             (absolute) filepath for current hdf5 container
         """
         iterations = []
+        if iteration is None:
+            iteration = list(handle["/data"].keys())[0]
         encoding = handle.attrs["iterationEncoding"].decode()
         if "groupBased" in encoding:
             iterations = list(handle["/data"].keys())
-            mylog.info("open_pmd - found {} iterations in file".format(len(iterations)))
+            mylog.info("Found {} iterations in file".format(len(iterations)))
         elif "fileBased" in encoding:
             itformat = handle.attrs["iterationFormat"].decode().split("/")[-1]
             regex = "^" + itformat.replace("%T", "[0-9]+") + "$"
             if path is "":
-                mylog.warning("open_pmd - For file based iterations, please use absolute file paths!")
+                mylog.warning("For file based iterations, please use absolute file paths!")
                 pass
             for filename in listdir(path):
                 if match(regex, filename):
                     iterations.append(filename)
-            mylog.info("open_pmd - found {} iterations in directory".format(len(iterations)))
+            mylog.info("Found {} iterations in directory".format(len(iterations)))
 
         if len(iterations) == 0:
-            mylog.warning("open_pmd - no iterations found!")
+            mylog.warning("No iterations found!")
         if "groupBased" in encoding and len(iterations) > 1:
-            mylog.warning("open_pmd - only choose to load one iteration ({})".format(list(handle["/data"].keys())[0]))
+            mylog.warning("Only chose to load one iteration ({})".format(iteration))
 
-        self.base_path = "/data/{}/".format(list(handle["/data"].keys())[0])
+        self.base_path = "/data/{}/".format(iteration)
         self.meshes_path = self._handle["/"].attrs["meshesPath"].decode()
+        try:
+            handle[self.base_path + self.meshes_path]
+        except(KeyError):
+            if self.standard_version <= StrictVersion("1.0.1"):
```

Comment:
to be honest, this will first be defined in `1.1.0` in the next weeks.
https://github.com/openPMD/openPMD-standard/issues/143